### ISSUE TITLE
W1514 unspecified-encoding false positive when mode is specified using a variable and it can be 'b' (binary mode)

### ISSUE THREAD

**Issue by mitesh-raval** [pylint-dev/pylint#10201]
### Bug description

```python
@contextmanager
def open_file_seek_zero(some_str, mode):
    ...
    with open(file_name, mode) as f:        # <===   this line is reported for W1514
        f.seek(0)
        yield f

============ above function usage ============
def some_other_func:
    with open_file_seek_zero(some_str, "r+b") as f:
        ...
==============================================
```

### Configuration

```ini

```

### Command used

```shell
pylint --rc=.pylintrc file_ops.py (made up file name)
```

### Pylint output

```python
************* Module file_ops
file_ops.py:29:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```

### Expected behavior

When file opening mode is not clearly discernable, unspecified-encoding should not be reported.

If I disable it in rc, I won't catch genuine errors.

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.10.5
```

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. I agree that when we don't know because we cannot infer we shouldn't raise a message.
